frontend/src/screens/WaitingRoom.js

```diff
@@ -1,51 +1,30 @@
-import React, { Component } from "react";
+import React, { useContext } from "react";
 import Typography from "@material-ui/core/Typography";
 import InputName from "../components/InputName";
 import WaitingPlayers from "../components/WaitingPlayers";
 import Button from "@material-ui/core/Button";
 import { MyContext } from "../infra/MyContext.js";
 
-const MOCK_playerList = ["Elisha", "Lahav", "Aviad"];
-class WaitingRoom extends Component {
-  constructor() {
-    super();
-    this.state = {
-      isUserEntered: false,
-      playerList: MOCK_playerList,
-    };
-  }
+const WaitingRoom = () => {
+  const context = useContext(MyContext);
 
-  render() {
-    return (
-      <div className="waiting-room">
-        <MyContext.Consumer>
-          {(context) => {
-            return (
-              <div>
-                <Typography variant="h1" component="h2" gutterBottom>
-                  Waiting Room - RoomId: {context.state.currentRoom}
-                </Typography>
-                {!context.state.isUserEnteredName ? (
-                  <>
-                    <InputName />
-                  </>
-                ) : (
-                  <></>
-                )}
+  return (
+    <div className="waiting-room">
+      <div>
+        <Typography variant="h1" component="h2" gutterBottom>
+          Waiting Room - RoomId: {context.state.currentRoom}
+        </Typography>
+        {!context.state.isUserEnteredName ? <EnterName /> : null}
```

Comment:
`!context.state.isUserEnteredName && <EnterName /> ` is not what you need?

Reply:
Great call
https://github.com/aviadch/Mafia/pull/7/commits/4cf0dab567dc11c0a4823e274851e3c481a18423